go/adbc/driver/flightsql/flightsql_connection.go

```diff
@@ -95,6 +97,115 @@ func doGet(ctx context.Context, cl *flightsql.Client, endpoint *flight.FlightEnd
 	return nil, err
 }
 
+func (c *cnxn) getSessionOptions(ctx context.Context) (map[string]interface{}, error) {
+	ctx = metadata.NewOutgoingContext(ctx, c.hdrs)
+	var header, trailer metadata.MD
+	rawOptions, err := c.cl.GetSessionOptions(ctx, &flight.GetSessionOptionsRequest{}, grpc.Header(&header), grpc.Trailer(&trailer), c.timeouts)
+	if err != nil {
+		// We're going to make a bit of a concession to backwards compatibility
+		// here and ignore UNIMPLEMENTED or INVALID_ARGUMENT
```

Comment:
UNIMPLEMENTED i understand, by why ignore INVALID_ARGUMENT?

Reply:
because that's what dremio and others send

Reply:
ugh, well that's annoying